src/test/java/com/ryanharter/autoparcel/AutoParcelExtensionTest.java

```diff
@@ -1,21 +1,82 @@
 package com.ryanharter.autoparcel;
 
+import android.os.Parcelable;
+import autovalue.shaded.com.google.common.auto.common.MoreElements;
+import com.google.auto.common.MoreTypes;
+import com.google.auto.value.AutoValueExtension;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.LinkedHashMultimap;
+import com.google.common.collect.SetMultimap;
+import com.google.testing.compile.CompilationRule;
+import com.ryanharter.autoparcel.util.TestMessager;
+import com.ryanharter.autoparcel.util.TestProcessingEnvironment;
 import com.squareup.javapoet.ClassName;
 import com.squareup.javapoet.FieldSpec;
 import com.squareup.javapoet.MethodSpec;
 import com.squareup.javapoet.TypeName;
-import java.util.HashMap;
+
+import java.util.Arrays;
+import java.util.Iterator;
 import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
 import java.util.Map;
+import java.util.Set;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
 import javax.lang.model.element.Modifier;
+import javax.lang.model.element.PackageElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.ElementFilter;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
 
 import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
 
 public class AutoParcelExtensionTest {
 
+  public @Rule CompilationRule rule = new CompilationRule();
+
   AutoParcelExtension extension = new AutoParcelExtension();
 
+  private Elements elements;
+  private Types types;
+  private ProcessingEnvironment processingEnvironment;
+
+  @Before public void setup() {
+    Messager messager = new TestMessager();
+    elements = rule.getElements();
+    types = rule.getTypes();
+    processingEnvironment = new TestProcessingEnvironment(messager, elements, types);
+  }
+
+  @Test(expected = AutoParcelException.class)
+  public void throwsForNonParcelableProperty() throws Exception {
+    TypeElement type = elements.getTypeElement(SampleTypeWithNonSerializable.class.getCanonicalName());
+    AutoValueExtension.Context context = createContext(type);
+
+    extension.generateClass(context, "Test_AnnotatedType", "SampleTypeWithNonSerializable", true);
+    fail();
```

Comment:
Also, FWIW, we tend to avoid both `expected=` and the `ExpectedException` JUnit rule in favor of try/catch because it lets you assert on the exception type (and potential causes) as well as the contents of its message. It also lets you do multiple try/catch assertions in a single test whereas both of the aforementioned approaches only allow one.

That said, I don't think this really needs changed unless there's going to be _a lot_ of exception-based testing in multiple tests.